src/tests/performance/get_trans.rs

```diff
@@ -0,0 +1,241 @@
+
+            // If there is address for our thread in the pool we can use it.
+            // Otherwise we'll not set bound_addr and use local IP addr (127.0.0.1).
+            let ip = if let Some(ip_addr) = ips.pop() {
+                SocketAddr::new(
+                    IpAddr::V4(Ipv4Addr::from_str(ip_addr).unwrap()),
+                    CONNECTION_PORT,
+                )
+            } else {
+                "127.0.0.1:0".parse().unwrap()
+            };
+
+            socket.bind(ip).expect("unable to bind to socket");
+            synth_sockets.push(socket);
+        }
+
+        // setup metrics recorder
+        let test_metrics = TestMetrics::default();
+        // clear metrics and register metrics
+        metrics::register_histogram!(METRIC_LATENCY);
+
+        // Wait for correct state and account data.
+        wait_for_state(&node.rpc_url(), "proposing".into()).await;
+        let account_data =
+            wait_for_account_data(&node.rpc_url(), TEST_ACCOUNT, EXPECTED_RESULT_TIMEOUT)
+                .await
+                .expect("unable to get account data");
+
+        // Get transaction info by rpc to put in cache.
+        let tx = account_data.result.account_data.previous_transaction;
+        let _ = get_transaction_info(&node.rpc_url(), tx.clone())
+            .await
+            .expect("unable to get transaction info");
+
+        let mut tx_hash = [0u8; TX_HASH_LEN];
+        hex::decode_to_slice(&tx, &mut tx_hash as &mut [u8])
+            .expect("unable to decode transaction hash");
+
+        let mut synth_handles = Vec::with_capacity(synth_count);
+        let test_start = tokio::time::Instant::now();
+
+        for socket in synth_sockets {
+            synth_handles.push(tokio::spawn(simulate_peer(node_addr, socket, tx_hash)));
+        }
+
+        // wait for peers to complete
+        for handle in synth_handles {
+            let _ = handle.await;
+        }
+
+        let time_taken_secs = test_start.elapsed().as_secs_f64();
+
+        let snapshot = test_metrics.take_snapshot();
+        if let Some(latencies) = snapshot.construct_histogram(METRIC_LATENCY) {
+            if latencies.entries() >= 1 {
+                // add stats to table display
+                table.add_row(RequestStats::new(
+                    synth_count as u16,
+                    REQUESTS,
+                    latencies,
+                    time_taken_secs,
+                ));
+            }
+        }
+
+        node.stop().unwrap();
+    }
+
+    // Display results table
+    println!("\r\n{}", table);
+}
+
+async fn simulate_peer(node_addr: SocketAddr, socket: TcpSocket, tx_hash: [u8; TX_HASH_LEN]) {
+    let mut synth_node = SyntheticNode::new(&Default::default()).await;
+
+    // Establish peer connection
+    synth_node
+        .connect_from(node_addr, socket)
+        .await
+        .expect("unable to connect to node");
+
+    let mut seq: u32 = 1;
+
+    for _ in 0..REQUESTS {

```

Comment:
why not `for seq in 1..=REQUESTS`?

Reply:
Changed.

Reply:
ah, so it can start from `0` as well? cool, even simpler

Reply:
Yes, it can, as in rippled code (PeerImpl.cpp:2680) it's just simple:
```
if (packet.has_seq())
     reply.set_seq(packet.seq());
```
so we can use zero. Initially used one as it was in c007_TM_GET_OBJECT_BY_HASH_get_transaction_by_hash packet construction code. And as there is only one packet constructed it does not matter if there is one, zero or 666 :)